### ISSUE TITLE
Workshop

### ISSUE THREAD

**Issue by NataliaLKB** [node-girls/hackday#3]
# What is Node?

This was really well received and a really important part of the day, is there anything we want to improve/change?
**Who is responsible for this section and will deliver it on the day?**
# Introduction to Node

Also really crucial. Can we tweak this to go more in depth into best practises (like error handling in the examples - this was requested by an attendee on the day) 
**Who is responsible for this section and will deliver it on the day?**
# Build a CMS

Needs to be simplified and less modular. I propose that we keep a branch similar to now that can be used as a reference for later learning.
**Who is responsible for this section and will deliver it on the day?**

**Comment by heron2014** [comment]
# Proposition of the workshop build

Myself and @minaorangina have an idea how to do the flow of the workshop. We believe that the whole example could be divided into small chunks of code which will be separated into branches (could be even 10 of them, the idea is to not move to the next step before submitting previous one). Every chunk of code should be no more than 10 lines. Each chunk of code will be referenced by README tutorial which will consist of : What? Why? How? related to that particular chunk. In this tutorial should be explanation what we are trying to achieve, why we doing the way we doing and how to resolve it with references where they can find help (sort of tips). In tutorial won't be an exact code but only direction how to solve the problem. 

Once they write their own solution to that chunk of example , they can move to another branch with another step, which follow the same logic meaning reference to the tutorial with explanation for that particular chunk.

That way all steps of the whole example will be referenced and explained in more depth, which may reduce confusion and lots of questions. 

Myself and @minaorangina, we are more than happy to do that on our hackday. If you guys think this is a good idea of course? and of course anyone can join us :) 

Please share your thoughts. I hope I explained that well.

**Comment by NataliaLKB** [comment]
Love the idea! 

Of course this isn't necessary, but will the attendees have left with something to show others?

**Comment by NataliaLKB** [comment]
On another read, is this for assisting the afternoon, or is this to replace the existing workshop completely?

**Comment by heron2014** [comment]
My initial thought was to replace the whole thing except maybe short intro to Node and then jump in to the workshop which will consist with all in depth explanation.

**Comment by NataliaLKB** [comment]
Makes sense, three more questions:
- Will the attendees have left with something to show others?
- If not, will the code examples relate to each other? 
- If so, how?

**Comment by heron2014** [comment]
The attendees will end up with the working example. (That's the goal). Plus on top of that they should understand the whole steps because of the README tutorial.

**Comment by heron2014** [comment]
All the chunks of code will lead to our finished working example.

It's gonna be CMS example but divided into chunks.

**Comment by NataliaLKB** [comment]
Oh okay! That makes more sense. I like this idea so far :+1:

**Comment by minaorangina** [comment]
Yeah the thinking was to deal with the whole issue of people feeling scared to admit if they didn't understand/were lost.  Though feedback for the lecture style was positive, the more "lecture-y" it feels, the less likely people are to ask questions.  

Also, having small, well documented chunks would probably make it easier to keep up, plus the added bonus of helping mentors be familiar with the material and enable them to help people out more effectively.

Would probably still be wise to have the CMS _**and**_ the simple Hello World server, and have both of them broken up into practical chunks?  A CMS is conceptually a bit of a step-up, so it's important people still actually get the basics of what a server actually is first.

Can think about this all on the hackday though :relaxed:

**Comment by iteles** [comment]
Such a great idea, this is always how I function best.

**Comment by NataliaLKB** [comment]
Agreed! You guys should definitely lead on the hackday for the workshop.

**Comment by nikhilaravi** [comment]
@heron2014 @minaorangina so just to check you'll be integrating both workshops into one and then splitting it up into continuous chunks building up to a full working cms at the end?

**Comment by minaorangina** [comment]
@nikhilaravi Kinda.  The cms will definitely be broken into chunks to build it up gradually, but I reckon it's definitely still worth having the simple Hello World server, without any extra complexity.  

But the idea was to see if we could move away from the pure lecture style and get people to build their Hello World server more in groups, with a personal mentor helping out, to encourage questions and avoid the situation where people were too timid to "expose themselves" when we were doing it all together.

Probably still worth having a presence of some sort at the front to explain things, but most of the actual building could be in groups where people can chat with their team members and their mentor if they need help (and not sit in silence pretending to follow)

Plus added bonus of the mentors having more of a defined role this time :+1:
